From 89415c01b730e3a1f6f66402b5c439fb04bda201 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Tue, 7 Dec 2021 22:22:53 +0100
Subject: [PATCH] Fix build with musl libc

canonicalize_file_name() is a GNU extension not supported by musl libc.
Use realpath() from POSIX instead.
---
 src/app-indicator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/app-indicator.c b/src/app-indicator.c
index a0e1553..cb0086f 100644
--- a/src/app-indicator.c
+++ b/src/app-indicator.c
@@ -2081,7 +2081,7 @@ append_snap_prefix (const gchar *path)
 	g_autofree gchar *canon_path = NULL;
 
 	if (snap != NULL && path != NULL) {
-		canon_path = canonicalize_file_name(path);
+		canon_path = realpath(path, NULL);
 
 		if (g_str_has_prefix (canon_path, "/tmp/")) {
 			g_warning ("Using '/tmp' paths in SNAP environment will lead to unreadable resources");
-- 
2.34.1

